### ISSUE TITLE
memprof: unexpected "move:-9223372036854775808"

### ISSUE THREAD

**Issue by olegrok** [tarantool/tarantool#5777]
```
../tarantool/src/tarantool --version
Tarantool 2.8.0-47-g7722a81c6
Target: Darwin-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=ON
Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
C_FLAGS: -Wno-unknown-pragmas -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-gnu-alignof-expression
CXX_FLAGS: -Wno-unknown-pragmas -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-invalid-offsetof -Wno-gnu-alignof-expression
```

```
ALLOCATIONS
INTERNAL: 8696552	826333470	0
@/Users/o.babin/Projects/tdg/common/document/comparators.lua:171, line 194: 400000	20800000	0
@builtin/buffer.lua:93, line 95: 400000	9600000	0
@test/bench/repository/unflatten.lua:167, line 168: 400000	22400000	0
@/Users/o.babin/Projects/tdg/common/document.lua:451, line 461: 300000	15200000	0
@builtin/msgpackffi.lua:296, line 300: 200001	4800039	0
@/Users/o.babin/Projects/tdg/common/repository.lua:136, line 159: 200000	11200000	0
@/Users/o.babin/Projects/tdg/common/document.lua:464, line 465: 200000	11200000	0
@test/bench/repository/unflatten.lua:167, line 169: 200000	9600000	0
@/Users/o.babin/Projects/tdg/common/document/comparators.lua:171, line 173: 100000	6400000	0
@/Users/o.babin/Projects/tdg/common/document/comparators.lua:171, line 172: 100000	6400000	0
@/Users/o.babin/Projects/tdg/common/repository.lua:228, line 258: 100000	9600000	0
@/Users/o.babin/Projects/tdg/common/model_flatten.lua:407, line 415: 100000	6400000	0
@/Users/o.babin/Projects/tdg/common/repository.lua:63, line 74: 100000	6400000	0
move:-9223372036854775808, line 0: 100000	2400000	0                  <<< Looks like a bug
@/Users/o.babin/Projects/tdg/common/repository.lua:136, line 138: 100000	6400000	0
```

Also I've sent my binary file with profiler results to @Buristan

**Comment by Buristan** [comment]
Fast functions defined via `LJLIB_LUA` macro is Lua functions, but there is no information about their `firstline` in `GCproto` structure (is set to -1).
Reproducer:
```lua
local t = {}
for i=1,100 do
        t[i] = i
end
local newt = {}
misc.memprof.start("/tmp/memprof_unexpected_move")
table.move(t, 1, 100, 1, newt)
misc.memprof.stop()
```
The old profiler report (before commit tarantool/luajit@cdf98db6f5b8dd5c51b1070ab309de823681763c) looks like the following:

```
ALLOCATIONS
INTERNAL: 2     540     0
move:-9223372036854775808, line 0: 1    24      0

REALLOCATIONS
move:-9223372036854775808, line 0: 6    2064    1056
        Overrides:
                move:-9223372036854775808, line 0

INTERNAL: 1     512     256

DEALLOCATIONS
INTERNAL: 1     0       48
        Overrides:
                INTERNAL

```
The new profiler report looks like the following:
```
ALLOCATIONS
INTERNAL: 2 events      +540 bytes      -0 bytes
move:0: 1 events        +24 bytes       -0 bytes

REALLOCATIONS
move:0: 6 events        +2064 bytes     -1056 bytes
        Overrides:
                move:0

INTERNAL: 1 events      +512 bytes      -256 bytes

DEALLOCATIONS
INTERNAL: 1 events      +0 bytes        -48 bytes
        Overrides:
                INTERNAL
```

**Comment by fckxorg** [comment]
This issue was, in fact, resolved by commit tarantool/luajit@cdf98db6f5b8dd5c51b1070ab309de823681763c, since now line numbers come from the location table instead of symtab. In the case of an allocation in a fast function, memprof dumps it as a Lua function allocation if a Lua function is the callee of that fast function, so the line number is normal. Otherwise, it dumps the fast function as a C function, so no issues with line number are even possible.
